packages/cardano-services-client/src/WebSocket.ts

```diff
@@ -145,16 +230,37 @@ export class CardanoWsClient extends WsProvider {
 
     const ws = (this.ws = new WebSocket(this.url));
 
-    // eslint-disable-next-line sonarjs/cognitive-complexity, complexity
+    // eslint-disable-next-line sonarjs/cognitive-complexity, complexity, max-statements
     ws.onmessage = (event) => {
       try {
         if (typeof event.data !== 'string') throw new Error('Unexpected data from WebSocket ');
 
         const message = fromSerializableObject<WSMessage>(JSON.parse(event.data));
-        const { clientId, networkInfo } = message;
+        const { clientId, networkInfo, responseTo, syncing, transactions, utxos } = message;
```

Comment:
Are we pushing all transactions to all clients now? This might result in higher network load :thinking:

Reply:
No, this is the client. Here `transactions` has a value only when a new tx relevant for this client is detected and sent by the server.

Reply:
Why is `isTxRelevant` function in the client then? :thinking:

Reply:
Both server and client use it and the `cardano-services` already imports the `cardano-services-client`.

Why the client uses it?
`CardanoWsClient` internally stores all the transactions to serve subsequent `transactionsByAddresses` calls and it does this for all the addresses it is called.
Imagine a scenario where it is called for address "a", next for address "b" last for address "c"; the third time it is called it knows all the transactions for all the three addresses, but only the ones relevant for the call must be provided in output.